chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -343,10 +451,46 @@ where
             })
             .collect::<Result<Vec<_>, _>>()?;
 
-        if !tx_undo.is_empty() {
+        // delete delegation if the balance is 0 and the pool has been decommissioned
+        let delete_delegation_undo = match check_for_delegation_cleanup {
+            Some(delegation_id) => {
+                let accounting_view = self.accounting_delta_adapter.accounting_delta();
+                let delegation_balance =
+                    accounting_view.get_delegation_balance(delegation_id)?.ok_or(
+                        ConnectTransactionError::DelegationDataNotFound(delegation_id),
+                    )?;
+                if delegation_balance == Amount::ZERO {
```

Comment:
Shouldn't we run this same check when a pool is decommissioned, for all the delegations?

Reply:
good point, thanks

Reply:
created an issue for that 
https://github.com/mintlayer/mintlayer-core/issues/909